### DISCUSSION TITLE
How to adjust the timezone?

### DISCUSSION THREAD

**Discussion by sleeplessKomodo** [xdpirate/calorific#2] [Q&A] [answered]
How to adjust the timezone? Currently loving the application, but having to manually adjust the time for each meal is a little annoying. 

I tried changing the "TZ" variable for the docker container, to no avail. My timezone is "America/Toronto".

**Comment by xdpirate** [comment]
Hey, thanks! If you're using the Docker container, I'm not sure how you would accomplish this right now, as I'm not that well-versed with configuring Docker.

If you're not, you can put `date_default_timezone_set('America/Toronto');` on a separate line in `credentials.php`.

I will look into adding support for this as an option for you to adjust in the next release, so it can be adjusted through the UI :) This should be rather trivial to accomplish and I will have a look at it soon.

**Comment by xdpirate** [comment]
Actually... The more I look at the code, the more I'm convinced this is a configuration error on your end - I seem to don't use any time zones at all and always rely on the currently configured system time. Is your Docker/MySQL/PHP/Apache time or system time offset from actual time? When saving an entry, I rely on MySQL's [`NOW()` function](https://www.w3schools.com/sql/func_mysql_now.asp) to provide current time. This gets the time from the server MySQL is running on and saves it to the database. If that time is offset actual time, there is a configuration error on the server somewhere 🤔

**Comment by sleeplessKomodo** [comment, marked as answer]
Ahh, I see now. I changed added the TZ variable to the mysql database container for calorific, and that seemed to do the trick! Thanks for the guidance!

**Comment by xdpirate** [comment]
Great to hear :)

**Comment by xdpirate** [comment]
For anyone coming across this in the future, a setting has been added to use an hour offset across the entire application, in case you're hosting Calorific in an environment where you're unable to change the server time.
